### PR TITLE
Update the recommended version upgrade path

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Specify that prior to 8.4, the Helm chart upgrade path is to the latest patch of the next **minor** version.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/setup/upgrade.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Tabs from "@theme/Tabs";
 import TabItem from "@theme/TabItem";
 
 :::note
-When upgrading to a new version of the Camunda 8 Helm charts, we recommend updating to the **latest patch** release of the **next major version**.
+When upgrading to a new version of the Camunda 8 Helm charts, we recommend updating to the **latest patch** release of the next **major** version of the chart.
```
Comment (aabouzaid) [inline docs/self-managed/setup/upgrade.md:12]: @conceptualshark What do you think about giving an example here?
So, adding something like:

> For example, If the current Helm chart version is 10.x.x and the latest next major version is 11.0.1, then it's recommended to upgrade to 11.0.1, not 11.0.0.

But that's just a suggestion and it's up to you.

**Review by aabouzaid — DISMISSED** [review]

File: versioned_docs/version-8.2/self-managed/platform-deployment/helm-kubernetes/upgrade.md [hunk 2]
```diff
@@ -5,14 +5,12 @@ sidebar_label: "Upgrade"
...
-
-:::caution
-
-Ensure to review the [instructions for specific version](#version-update-instructions) before starting the actual upgrade.
-
+:::note
+When upgrading to a new version of the Camunda 8 Helm charts, we recommend updating to the **latest patch** release of the next **minor** version of the chart.
 :::
 
+To upgrade to a more recent version of the Camunda Helm charts, there are certain things you need to keep in mind. Review the [instructions for a specific version](#version-update-instructions) before starting the actual upgrade.
```
Comment (mesellings) [inline versioned_docs/version-8.2/self-managed/platform-deployment/helm-kubernetes/upgrade.md:12]:
```suggestion
Before upgrading to a more recent version of the Camunda Helm charts, you should review the [instructions for a specific version](#version-update-instructions).
```
Just a rewording suggestion for conciseness - wasn't sure whetehr to use "should" or "must" either?

**Review by mesellings — DISMISSED** [review]
Lgtm - only a non-blocking suggestion for rewording 👍

**Comment by conceptualshark** [comment]
@mesellings tagging you for a quick re-review on this (old 😱) PR, just some rewording + adding a suggested example.

**Review by mesellings — APPROVED** [review]
Lgtm, just the rewording suggestion I still feel could be useful, as it still feels like that note about upgrading could be less wordy? But happy to approve anyway 👍

**Pull request merged**